site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -58,7 +59,7 @@ const CreateWorkspacePage: FC = () => {
         templateName={templateName}
         templates={templates}
         selectedTemplate={selectedTemplate}
-        templateParameters={templateParameters}
+        templateParameters={orderedTemplateParameters(templateParameters)}
```

Comment:
Suggestion: Could be cleaner to pass two separate, mutable and immutable parameter properties here.

I'm thinking about how they're being used, and e.g. needing to filter to check length etc.

Reply:
This is a trick to let Formik properly process the request body correctly as all parameters are linked with a single response field `rich_parameter_values[]`.

I'm not sure how to link two different arrays (`mutableTemplateParameters` and `immutableTemplateParameter`) into a single `rich_parameter_values[]` without [calculating indices](https://github.com/coder/coder/blob/main/site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx#L281-L295).

Reply:
Ahaa, I see. Then it makes more sense why it's the way it is. Perhaps not worth it 👍🏻